### PR TITLE
[Opt](limit) Avoid copy in set_num_rows implementation

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

if the column has no extra reference, we could assume_mutable and just resize it. no need to copy.

perf:
```
Run on (96 X 3054.64 MHz CPU s)
CPU Caches:
  L1 Data 32 KiB (x48)
  L1 Instruction 32 KiB (x48)
  L2 Unified 1024 KiB (x48)
  L3 Unified 36608 KiB (x2)
Load Average: 14.09, 16.44, 35.13
-----------------------------------------------------
Benchmark           Time             CPU   Iterations
-----------------------------------------------------
SHRINK     7.3495e+13 ns     44527988 ns            1
CUT        7.3496e+13 ns   1638929204 ns            1
```

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/columns/column.h [hunk 1]
```diff
@@ -185,12 +185,21 @@ class IColumn : public COW<IColumn> {
 
     /// Removes all elements outside of specified range.
     /// Is used in LIMIT operation, for example.
-    virtual Ptr cut(size_t start, size_t length) const {
+    virtual Ptr cut(size_t start, size_t length) const final {
```
Comment (yiguolei) [inline be/src/vec/columns/column.h:188]: add beut for these 2 methods

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]

**Pull request merged**